server/mcp_server.py

```
@@ -140,28 +176,93 @@ def _get_token() -> str:
+    Default request timeout is 30s (configured on the shared httpx client).
+    Callers that need a different deadline wrap with `async with asyncio.timeout(N)`.
+    """
+    if not path.startswith("/") or path.startswith("//"):
         raise ValueError(f"Path must be a relative path starting with /: {path}")
+    # Defense-in-depth: httpx.URL collapses `..` segments, so an unencoded
+    # path traversal would escape the intended prefix even though the path
+    # "starts with /". Callers must URL-encode user input (see
+    # urllib.parse.quote in tools_gated/dispatch); reject anything that still
+    # contains a parent-segment so a single missed call site can't leak.
+    if "/.." in path or path.endswith("/..") or "%2e%2e" in path.lower():
+        raise ValueError("Path contains parent-directory segments")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)